### PR TITLE
Update readme's documentation

### PR DESCRIPTION
**Pull request by refnil**
- Update examples to recent version of nixpkgs
- Update obsolete sections following changes to niv
- Add information about the different ways of using niv sources

### CODE REVIEW

**Review by nmattia — APPROVED** [review]
Thanks, this is GOLD!

File: README.tpl.md [hunk 1]
```diff
@@ -302,6 +294,74 @@ netrc-file = /PATH/TO/.netrc
...
+let hello_src = sources.hello-src;
+    nixpkgs =  import sources.nixpkgs {};
+in nixpkgs.stdenv.mkDerivation {
+    pname = "hello";
+    version = "custom";
+    src = hello_src;
+}
+```
+
+:warning: If you have problems, consider using the outPath of the source
```
Comment (nmattia) [inline README.tpl.md:361]: does `:warning:` actually work? is that a GitHub flavored markdown feature?

Comment (refnil) [reply README.tpl.md:361]: The `:word:` syntax is for writing emoji in github markdow. `:warning:=`:warning: 
It is definitely not a basic markdown feature but other markdown variant integrate this feature too. I'm aware that gitlab's markdown has this feature.

Comment (nmattia) [reply README.tpl.md:361]: hah! of course.

**Pull request merged**